File: C_Programming/Competitive_Programming/CP_018.c
//////////////////////////////////////////////////////////////////
//
//  File Name    : Display_Non_Factors
//
//  Description  : This program accepts a number from the user
//                 and displays all the non-factors of the given
//                 number, excluding the number itself.
//
//  Example:
//      Input  : 12
//      Output : 5 7 8 9 10 11
//
//      Input  : 13
//      Output : 2 3 4 5 6 7 8 9 10 11 12
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

void NonFact(int iNo)
{
    int i = 0;

    for(i = 1; i < iNo; i++)
    {
        if(iNo % i != 0)
        {
            printf("%d\t", i);
        }
    }
}

int main()
{
    int iValue = 0;

    printf("Enter Number : \n");
    scanf("%d", &iValue);

    NonFact(iValue);

    return 0;
}

File: Java_Programming/Competitive_Programming/CP_018.java
//////////////////////////////////////////////////////////////////
//
//  File Name    : Check_Perfect_Number
//
//  Description  : This program accepts a number and
//                 checks whether the given number is
//                 a perfect number or not.
//
//  Example:
//      Input  : 6
//      Output : Perfect Number.
//
//////////////////////////////////////////////////////////////////

class Logic
{
    void checkPerfect(int iNum)
    {
        int iCnt = 0, iSum = 0;

        for(iCnt = 1; iCnt <= iNum / 2; iCnt++)
        {
            if(iNum % iCnt == 0)
            {
                iSum = iSum + iCnt;
            }
        }
        if(iSum == iNum)
        {
            System.out.println("Perfect Number.");
        }
        else
        {
            System.out.println("Not a Perfect Number.");
        }
    }
}

class CP_018
{
    public static void main(String A[])
    {
        Logic lobj = new Logic();

        lobj.checkPerfect(6);
    }
}